pkg/querier/frontend/frontend.go

```diff
@@ -139,7 +143,20 @@ func (f *Frontend) Handler() http.Handler {
 }
 
 func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
+	startTime := time.Now()
 	resp, err := f.roundTripper.RoundTrip(r)
+	queryResponseTime := time.Now().Sub(startTime)
+
+	userID, err := user.ExtractOrgID(r.Context())
+	if err != nil {
+		server.WriteError(w, err)
+		return
+	}
+
+	if f.cfg.LogQueriesLongerThan > 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
+		level.Warn(f.log).Log("msg", "slow query", "orgID", userID, "url", fmt.Sprintf("http://%s", r.Host+r.RequestURI), "time-taken", queryResponseTime.String())
```

Comment:
I'm a bit dubious about the `Warn` level. A warning or error would mean that it's actionable. This log is not necessarily actionable (think about a multi-tenant system and the query is implicitly slow because complex and/or spanning over a large time range). I would see `Info` more appropriate here.

Reply:
Please also replace `"orgID"` with `"user"`.

Reply:
:+1: to both. The info level logging is pretty much non-existent for the frontend and this won't get lost afaics.

Reply:
Actually, I prefer `orgID/tenant` over user. And `orgID` is the majority term in use right now. Maybe just variable from `userID` to `orgID`?

Reply:
`tenant` looks the right naming to me, but I've mentioned `user` to keep consistency with the other logs / metrics (it's usually called "user"). I let you @gouthamve say the last word on this.

Reply:
Argh, I just did a quick grep and we use `org_id` and `user` interchangeably everywhere. I think it's best to standardise towards `org_id` rather than user, see https://github.com/cortexproject/cortex/blob/a5ecafc0782b4c480c3902f228e3409a647d82cb/pkg/util/log.go#L119-L124

Also, sadly tenant is not used anywhere.

Reply:
yeah, that makes sense to standardize towards `org_id` everywhere.

Done, updated

Reply:
Thanks @gouthamve